core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -338,7 +355,12 @@ public void innerWriteTo(StreamOutput out) throws IOException {
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startArray(name);
+            if (params.paramAsBoolean(RestSearchAction.TYPED_KEYS_PARAM, false)) {
+                // Concatenates the type and the name of the aggregation (ex: top_hits#foo)
+                builder.startArray(String.join(InternalAggregation.TYPED_KEYS_DELIMITER, getType(), getName()));
```

Comment:
why would we want to use the same parameter but not the same delimiter? aren't we making things more complicated than they are?